### PR TITLE
RepoProfilesCheck: check for banned&deprecated profile EAPIs

### PR DESCRIPTION
**Pull request by mgorny**
CC @radhermit

### CODE REVIEW

**Comment by mgorny** [comment]
Also CC @ulm

**Comment by mgorny** [comment]
(the CI failure is expected, I need to make a new pkgcore release)

File: src/pkgcheck/checks/profiles.py [hunk 1]
```diff
@@ -302,6 +302,35 @@ def desc(self):
...
+
+    def __init__(self, profile, eapi, **kwargs):
+        super().__init__()
+        self.profile = profile
+        self.eapi = str(eapi)
+
+    @property
+    def desc(self):
+        return (
+            f"{self.profile!r} profile is using {self._type} EAPI {self.eapi}"
```
Comment (radhermit) [inline src/pkgcheck/checks/profiles.py:318]: This doesn't need to be on a separate line.

File: src/pkgcheck/checks/profiles.py [hunk 2]
```diff
@@ -302,6 +302,35 @@ def desc(self):
         )
 
 
+class _ProfileEapiResult(results.ProfilesResult):
+    """Generic profile EAPI result."""
+
+    _type = None
+
+    def __init__(self, profile, eapi, **kwargs):
```
Comment (radhermit) [inline src/pkgcheck/checks/profiles.py:310]: I'd drop the ``**kwargs`` usage since it's not going to be used.

**Comment by mgorny** [comment]
Done and done!

**Pull request merged**